### PR TITLE
Milestone 2 - Add interactivity to App.

### PR DESCRIPTION
**Pull request by KingsleyIbe**
In this milestone, I implemented the following:

- Added new JavaScript files and import them as a module:
- Add event listener to the checkbox (change).
- Update items object's value for completed key upon user actions.
- Store the updated array of items in local storage, so the user gets the correct list values after the page reloads.

### CODE REVIEW

File: src/ultils/status.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const updateItems = () => {
+  document.querySelectorAll('.list-item .checked').forEach((item) => {
+    item.addEventListener('change', (e) => {
+      const data = JSON.parse(localStorage.getItem('listitems')) || listItems
+      const filtered = data.filter((item) => item.index !== parseInt(e.target.dataset.id, 10))
+      const currentData = data.find((item) => item.index === parseInt(e.target.dataset.id, 10))
+      currentData.completed = !currentData.completed;
+      filtered.push(currentData);
+      // localStorage.setItem('listitems', JSON.stringify(filtered));
```
Comment (Dipeshtwis) [inline src/ultils/status.js]: - [ ] I would recommend removing the commented code to make the codebase cleaner

File: src/ultils/status.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        localStorage.setItem('CurrentData', JSON.stringify(currentData));    
+    });
+  });
+};
+
+const storedData = JSON.parse(localStorage.getItem('listItems') || 'null');
+if (storedData !== null) {
+  listItems.length = 0;
+  listItems.push(...storedData);
```
Comment (Dipeshtwis) [inline src/ultils/status.js]: - [ ] I have changed the completed status of the task, after page reloads, it does not stay the same in UI. Please fix that.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @KingsleyIbe, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

- [ ] Please add instructions on how to clone, install and run this project, you should explain the instructions like if someone that never touched code before was trying to see and run your project :smile:

Hint:
```
$ git clone https://github.com/KingsleyIbe/to-do-list-app.git

$ cd to-do-list-app

and then installing the npm packages
$ npm i or npm install

and then to run the server
$ npm start

```

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KingsleyIbe** [comment]
Changes has be effected as recommended by the reviewer.

File: src/ultils/status.js (hunk 1 above)
Comment (henatan99) [reply src/ultils/status.js]: This issue has been addressed well. Nice job! :pray:

File: src/ultils/status.js (hunk 2 above)
Comment (henatan99) [reply src/ultils/status.js]: This issue has been addressed well. Nice job! :pray:

**Review by henatan99 — APPROVED** [review]
Hi @KingsleyIbe , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KingsleyIbe** [comment]
Thanks

**Pull request merged**